src/v/kafka/server/request_context.h

```diff
@@ -246,6 +246,16 @@ class request_context {
         return _conn->server().get_replica_selector();
     }
 
+    /// \ref kafka::server memory semaphore for the local shard
+    ssx::semaphore& memory_sem_local() const noexcept {
+        return _conn->server().container().local().memory();
+    }
+
+    /// \ref kafka::server fetch memory semaphore for the local shard
+    ssx::semaphore& memory_fetch_sem_local() const noexcept {
+        return _conn->server().container().local().memory_fetch_sem();
+    }
```

Comment:
can you add some emphasis to the comments that the local shard is for the _caller_? we don't really have any sort of formal rules around it, but generally working directly with the sharded type in these cases is nice because it forces the caller to explicitly write `.local()`. So alternatively another way to improve this interface other than a comment woudl be to do something like

```
ss::sharded<server>& server() return { _conn->server().container(); }
// at the caller
server().local().memory_fetch_sem();
```

Reply:
I like the way you have suggested. I did not do that because I thought the design was to expose only an explit API from `request_context`, but if it's not, then `ss::sharded<server>& server()` is a much better way

Reply:
it's all good. i think the design so far just didn't have a use case like this.